lib/storage/storage.go

```diff
@@ -673,7 +675,7 @@ var freeDiskSpaceLimitBytes uint64
 
 // IsReadOnly returns information is storage in read only mode
 func (s *Storage) IsReadOnly() bool {
-	return s.isReadOnly.Load()
+	return s.isReadOnly.Load() && s.cannotWrite.Load()
```

Comment:
isReadOnly can be altered in different places so seems to me like an overloaded variable if we do it like you propose. This way we make sure that it will stay readonly if we set it to it.